src/emulator/dataconnectEmulator.ts

```diff
@@ -30,7 +31,13 @@ export interface DataConnectBuildArgs {
 }
 
 export class DataConnectEmulator implements EmulatorInstance {
-  constructor(private args: DataConnectEmulatorArgs) {}
+  private emulatorClient: DataConnectEmulatorClient;
+  private origin: string;
+
+  constructor(private args: DataConnectEmulatorArgs) {
+    this.origin = args.listen[0].address + args.listen[0].port;
```

Comment:
Don't manually compute from address and ports, use `EmulatorRegistry.client` instead. `args.listen[0]` doesn't handle `0.0.0.0` and so on correctly, nor does it quote IPv6 properly.

https://github.com/firebase/firebase-tools/blob/1633d690e9ed84f9bfe68455fe76a3135681f0b7/src/emulator/registry.ts#L194

Reply:
oh this is a lot smarter, thanks